backend/orcid.py

```diff
@@ -144,133 +345,41 @@ def fetch_orcid_records(self, id, profile=None, use_doi=True):
-            citation_format = j('work-citation/work-citation-type')
-            print citation_format
-            bibtex = j('work-citation/citation')
-
-            if bibtex is not None:
-                try:
-                    entry = parse_bibtex(bibtex)
-
-                    if entry.get('author', []) == []:
-                        print "Warning: Skipping ORCID publication: no authors."
-                        print j('work-citation/citation')
-                    if not authors:
-                        authors = entry['author']
-                except ValueError:
-                    pass
-
-            affiliations = affiliate_author_with_orcid(ref_name, id, authors, initial_affiliations=affiliations)
-
-            authors = map(name_lookup_cache.lookup, authors)
-
-            if not authors:
-                print "No authors found, skipping"
+            # Extract information from ORCiD
+            if data_paper.skipped:
+                print ('%s is skipped due to incorrect metadata' % (data_paper))
+                ignored_papers.append(pub)
                 continue
 
-            # Create paper:
-            paper = BarePaper.create(title, authors, pubdate, 'VISIBLE', affiliations)
-
-            record = BareOaiRecord(
-                    source=orcid_oai_source,
-                    identifier=identifier,
-                    splash_url='http://orcid.org/'+id,
-                    pubtype=doctype)
-
-            paper.add_oairecord(record)
-            yield paper
+            yield self.create_paper(data_paper)
 
         if use_doi:
-            for metadata in crps.search_for_dois_incrementally('', {'orcid':id}):
-                try:
-                    paper = crps.save_doi_metadata(metadata)
-                    if paper:
-                        yield paper
-                except ValueError as e:
-                    print "Saving CrossRef record from ORCID failed: %s" % unicode(e)
-
-            # Now we add the DOIs found in the ORCID profile.
-            doi_metadata = fetch_dois(dois)
-            for metadata in doi_metadata:
-                try:
-                    authors = map(convert_to_name_pair, metadata['author'])
-                    affiliations = affiliate_author_with_orcid(ref_name, id, authors)
-                    paper = crps.save_doi_metadata(metadata, affiliations)
-                    if not paper:
-                        continue
-                    record = BareOaiRecord(
-                            source=orcid_oai_source,
-                            identifier='orcid:'+id+':'+metadata['DOI'],
-                            splash_url='http://orcid.org/'+id,
-                            pubtype=paper.doctype)
-                    paper.add_oairecord(record)
-                    yield paper
-                except (KeyError, ValueError, TypeError):
-                    pass
-
-orcid_oai_source, _ = OaiSource.objects.get_or_create(identifier='orcid',
```

Comment:
This should not be deleted! That's a global line of code (lying outside the previous function). It's also my fault, I should have put this at the beginning of the file instead.